### PR TITLE
add inline-const test for elided lifetimes being infer vars

### LINKED ISSUE #110934: inline const test for elided lifetimes being infer vars
There should be a test for `feature(inline_const)` ensuring that it does not have the same bug as #110931 and #110933, that `'_` inside of the const refers to `'static` instead of inferring a lifetime. The following code is what passes and should be added as a test:

```rust
#![feature(inline_const)]

fn main() {
    let my_usize = const {
        let a = 10_usize;
        let b: &'_ usize = &a;
        *b
    };
}
```

### PR DESCRIPTION
**Pull request by zirconium-n**
Fixes #110934

### CODE REVIEW

File: tests/ui/inline-const/elided-lifetime-being-infer-vars.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// run-pass
```
Comment (compiler-errors) [inline tests/ui/inline-const/elided-lifetime-being-infer-vars.rs:1]: this doesn't need to be run-pass

```suggestion
// check-pass
```

Comment (zirconium-n) [reply tests/ui/inline-const/elided-lifetime-being-infer-vars.rs]: Fixed

**Comment by compiler-errors** [comment]
please squash this into one commit. sorry, should've asked when i requested a change, but also i think it's typically good practice for small changes like this.

**Comment by zirconium-n** [comment]
Oh, yeah. that. I somehow forgot that too. :(

**Pull request merged**